sdk/synapse/synapse-spark/review/synapse-spark.api.md

```diff
@@ -0,0 +1,525 @@
+    detailed?: boolean;
+}
+
+// @public
+export type CreateSparkSessionResponse = SparkSession & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkSession;
+    };
+};
+
+// @public (undocumented)
+export type CreateSparkStatementOptions = OperationOptions;
+
+// @public
+export type CreateSparkStatementResponse = SparkStatement & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkStatement;
+    };
+};
+
+// @public
+export interface GetSparkBatchJobOptions extends coreHttp.OperationOptions {
+    detailed?: boolean;
+}
+
+// @public
+export type GetSparkBatchJobResponse = SparkBatchJob & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkBatchJob;
+    };
+};
+
+// @public
+export interface GetSparkSessionOptions extends coreHttp.OperationOptions {
+    detailed?: boolean;
+}
+
+// @public
+export type GetSparkSessionResponse = SparkSession & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkSession;
+    };
+};
+
+// @public (undocumented)
+export type GetSparkStatementOptions = OperationOptions;
+
+// @public
+export type GetSparkStatementResponse = SparkStatement & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkStatement;
+    };
+};
+
+// @public (undocumented)
+export interface ListPageSettings {
+    continuationToken?: string;
+}
+
+// @public
+export interface ListSparkBatchJobsOptions extends coreHttp.OperationOptions {
+    detailed?: boolean;
+    fromParam?: number;
+    size?: number;
+}
+
+// @public
+export type ListSparkBatchJobsResponse = SparkBatchJobCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkBatchJobCollection;
+    };
+};
+
+// @public
+export interface ListSparkSessionsOptions extends coreHttp.OperationOptions {
+    detailed?: boolean;
+    fromParam?: number;
+    size?: number;
+}
+
+// @public
+export type ListSparkSessionsResponse = SparkSessionCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkSessionCollection;
+    };
+};
+
+// @public (undocumented)
+export type ListSparkStatementsOptions = OperationOptions;
+
+// @public
+export type ListSparkStatementsResponse = SparkStatementCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkStatementCollection;
+    };
+};
+
+// @public
+export const logger: import("@azure/logger").AzureLogger;
+
+// @public (undocumented)
+export interface OperationResponse {
+    _response: HttpResponse;
+}
+
+export { PipelineOptions }
+
+// @public
+export type PluginCurrentState = "Preparation" | "ResourceAcquisition" | "Queued" | "Submission" | "Monitoring" | "Cleanup" | "Ended" | string;
```

Comment:
I think unioning this with `string` means TS will treat it as just a string without there being helpful intellisense. /cc @bterlson @willmtemple to correct me

Reply:
Yeah I think this is still true. For a while I was arguing with the TS team about making this do something useful, I will follow up with where that's at.

Reply:
This part is generated by autorest, because we set modelAsString as true using x-ms-enum extension at swagger, and it is recommended practice, that is for avioding breaking change when add/remove item from union type:
[https://github.com/Azure/azure-rest-api-specs/blob/fdf4bbfd7a73b28960d3a62490440345d6f2e8e3/specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/sparkJob.json#L857](https://github.com/Azure/azure-rest-api-specs/blob/fdf4bbfd7a73b28960d3a62490440345d6f2e8e3/specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/sparkJob.json#L857)  

And what can we do to fix it at JS/TS sdk level if it is a bad practice?




---
In reply to: [521608635](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r521608635) [](ancestors = 521608635)

Reply:
There isn't a great story here today, if you want it to be future proof then yes you can do this, but it effectively makes this type `string` in the type system, and your consumers will have to handle it exhaustively.

Reply:
Got it, i also send a feedback to autorest.typescript about it. Let's keep it as it is now~

---
In reply to: [524824167](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r524824167) [](ancestors = 524824167)